cassiopeia/solver/SpectralNeighborJoiningSolver.py

```diff
@@ -0,0 +1,318 @@
+
+        Returns:
+            DataFrame object of the lambda matrix, but similar in structure to
+                DistanceSolver's dissimilarity_map.
+        """
+
+        # get dissimilarity map and save it as private instance variable
+        self._similarity_map = super().get_dissimilarity_map(
+            cassiopeia_tree, layer
+        )
+
+        # prepare for first (pairwise) lambda matrix
+        N = self._similarity_map.shape[0]
+        node_names: np.ndarray = self._similarity_map.index.values
+        self.lambda_indices = [[i] for i in range(N)]
+
+        # generate the lambda matrix
+        lambda_matrix_arr = np.zeros([N, N])
+        for (j_idx, i_idx) in itertools.combinations(range(N), 2):
+
+            svd2_val = self._compute_svd2(
+                pair=(i_idx, j_idx), lambda_indices=self.lambda_indices
+            )
+
+            lambda_matrix_arr[i_idx, j_idx] = lambda_matrix_arr[
+                j_idx, i_idx
+            ] = svd2_val
+
+        np.fill_diagonal(lambda_matrix_arr, np.inf)
+
+        # convert array to dataframe
+        lambda_matrix_df = pd.DataFrame(
+            lambda_matrix_arr, index=node_names, columns=node_names
+        )
+
+        return lambda_matrix_df
+
+    def find_cherry(self, dissimilarity_map: np.ndarray) -> Tuple[int, int]:
+        """Finds a pair of samples to join into a cherry.
+
+        With dissimilarity_map being the lambda matrix, this method finds the
+            argmin pair of subsets of the lambda matrix.
+
+        Args:
+            dissimilarity_matrix: Lambda matrix
+
+        Returns:
+            A tuple of integers representing rows in the
+                dissimilarity matrix to join.
+        """
+
+        return np.unravel_index(
+            np.argmin(dissimilarity_map, axis=None), dissimilarity_map.shape
+        )
+
+    def _compute_svd2(
+        self, pair: Tuple[int, int], lambda_indices: List[List[int]]
+    ) -> float:
+        """Computes the second largest singular value of a pair of
+        subset's RA matrix.
+
+        Args:
+            pair: pair of indices i and j where i > j.
+            lambda_indices: the list of subsets for
+                which 'pair' refers to.
+
+        Returns:
+            The second largest singular value of the pair's RA matrix.
+        """
+        i_idx, j_idx = pair
+        i_sub, j_sub = lambda_indices[i_idx], lambda_indices[j_idx]
+
+        # get combined pair of subsets
+        a_subset = [*i_sub, *j_sub]
+
+        # get complement
+        a_comp = lambda_indices.copy()
+        a_comp.pop(i_idx)
+        a_comp.pop(j_idx)
+        a_comp_flat = list(itertools.chain.from_iterable(a_comp))
+
+        # reconstruct RA matrix
+        RA_matrix = self._similarity_map.values[np.ix_(a_subset, a_comp_flat)]
+
+        # get second largest SVD if available, first if not.
+        s = svd(RA_matrix, compute_uv=False, check_finite=False)
```

Comment:
It's better practice to use the form `module.function` -- so, I think you should avoid importing a specific function, if possible, and here you can call `scipy.linalg.svd`.